### PR TITLE
EDU-826 Correct logic for determining if doc editing is allowed and redirect if not allowed

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-826

### CODE REVIEW

File: src/server/document-controller.js [hunk 1]
```diff
@@ -105,6 +106,10 @@ class DocumentController {
       room = null;
     }
 
+    if (view === DOC_VIEW_QUERY_PARAM.edit && !canEditDocContent({ user, doc, room })) {
+      return res.redirect(routes.getDocUrl({ id: doc._id, slug: doc.slug, templateDocumentId }));
```
Comment (ahelmberger) [inline src/server/document-controller.js:110]: As the `view` param is not (and cannot be) "edit" here, we should also ditch any possible `templateDocumentId` from the URL here, I guess.

Comment (adrianaluchian) [reply src/server/document-controller.js:110]: I wanted to keep everything but the `view` param, but sure, I can also simplify.

Comment (adrianaluchian) [reply src/server/document-controller.js]: Done!

Comment (ahelmberger) [reply src/server/document-controller.js]: Cool, thanks!

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**